ocaml/xapi/storage_impl.ml

```diff
@@ -568,26 +573,29 @@ functor
           )
         in
         let failures =
-          List.fold_left
-            (fun acc dp ->
-              info "Attempting to destroy datapath dp:%s sr:%s vdi:%s" dp
-                (s_of_sr sr) (s_of_vdi vdi) ;
-              try
-                destroy_datapath_nolock context ~dbg ~dp ~sr ~vdi ~vm
-                  ~allow_leak:false ;
-                acc
-              with e -> e :: acc
-            )
-            [] dps
+          if fail_on_leaked && dps <> [] then
+            [failwith "Leaked Datapaths not allowed"]
```

Comment:
This raises an exception immediately rather than putting an exception _value_ in a list. I'd really just pull it outside `let failures =`, e.g.:

```
if fail_on_leaked && dps <> [] then
  failwith "Leaked datapaths not allowed" ;
let failures =
  ...
```